app/wama.web/Controllers/CheckInController.cs

```diff
@@ -27,10 +29,15 @@ public IActionResult Index()
             if (ModelState.IsValid && !string.IsNullOrWhiteSpace(memberId))
             {
                 var loginCredential = await _CheckInService.GetLogInCredentialAsync(memberId);
-
-                if (loginCredential == null)
+                var userAccount = await _UserAccountService.GetUserAccountAsync(memberId);
+                
+                if(userAccount == null || loginCredential == null)
```

Comment:
Here's my breakdown of the possible scenarios, hopefully it's helpful with breaking this down

loginCredential = lc
userAccount = ua


**lc == null** // No login credential

* Possible solutions
  * If **ua == null** (see below for how to deal with it)
  * Else
    * Most likely, the account hasn't been verified - (Talk to the front desk folks)
    * Something catastrophic has happened - (Don't worry about this)

**ua == nul**l // No user account
* No account with the member ID exists, create new account